src/app.ts

```diff
@@ -237,8 +250,20 @@ async function clearAndReloadSpidStrategy(
   // tslint:disable-next-line: no-object-mutation
   app._router.stack = app._router.stack.filter(not(isLoginRoute));
 
-  await registerLoginRoute(app);
-  log.info("Spid strategy re-initialization complete.");
+  try {
+    const newSpidStratedy = await defaultModule.loadSpidStrategy();
+    app.set(SPID_STRATEGY, newSpidStratedy);
```

Comment:
I'm confused, why you need to store the SPID strategy both in the app settings and in the awilix container?

Reply:
It's a workaround to avoid that the download and parsing of IDP metadata will be re-executed.  The value stored inside the awilix container is a Promise that returns the SPID strategy object. The value injected through awilix inside controllers execs the code only once, but if we read the value inside the container like `await container.resolve(SPID_STRATEGY)` the code will be re-executed.

Reply:
I don't like this - can you try to find a way to store it only once?